scripts/integration-test-all.sh

```diff
@@ -40,46 +44,66 @@ if [[ ! "$DATA_DIR" ]]; then
   exit 1
 fi
 
-
 # Compile gossamer
 echo "compiling gossamer"
 make build
 
+# PID array declaration
+arr=()
 
-# Run node with static blockchain database
-echo "starting gossamer node in background with http listener on $HOST_RPC"
+start_func() {
+  echo "starting gossamer node $i in background ..."
+  "$PWD"/bin/gossamer --port=$PORT"$i" --key=$KEY --datadir="$DATA_DIR$i" \
+    --rpc --rpchost=$IP_ADDR --rpcport=$RPC_PORT"$i" --rpcmods=system,author >"$DATA_DIR"/node"$i".log 2>&1 & disown
 
-$PWD/bin/gossamer --port=$PORT --key=$KEY --datadir=$DATA_DIR \
-                  --rpc --rpchost=$IP_ADDR --rpcport=$RPC_PORT --rpcmods=system,author &
-
-GOSSAMER_PID=$!
+  GOSSAMER_PID=$!
+  echo "started gossamer node, pid=$GOSSAMER_PID"
+  #add PID to array
+  arr+=("$GOSSAMER_PID")
+}
 
-echo "gossamer node pid=$GOSSAMER_PID"
+# Run node with static blockchain database
+# For loop N times
+for i in $(seq 1 "$QTD"); do
+  start_func "$i" #&# Put a function in the background
+  echo "sleeping $TIMEOUT seconds for startup"
+  sleep "$TIMEOUT"
+  echo "done sleeping"
+done
 
-echo "sleeping for startup"
-sleep 5
+echo "sleeping $TIMEOUT seconds before running tests ... "
+sleep "$TIMEOUT"
 echo "done sleeping"
 
 set +e
 
+if [[ -z $TEST || $TEST == "rpc" ]]; then
 
-if [[ -z $TEST || $TEST = "rpc" ]]; then
-
-GOSSAMER_INTEGRATION_TEST_MODE=$MODE GOSSAMER_NODE_HOST=$HOST_RPC go test ./tests/rpc/... -timeout=60s -v
+  for i in $(seq 1 "$QTD"); do
+    HOST_RPC=http://$IP_ADDR:$RPC_PORT"$i"
+    echo "going to test gossamer node $HOST_RPC ..."
+    GOSSAMER_INTEGRATION_TEST_MODE=$MODE GOSSAMER_NODE_HOST=$HOST_RPC go test ./tests/rpc/... -timeout=60s -v -count=1
 
-RPC_FAIL=$?
+    RPC_FAIL=$?
+  done
 
 fi
 
+stop_func() {
+  GOSSAMER_PID=$i
+  echo "shutting down gossamer node, pid=$GOSSAMER_PID"
```

Comment:
`shutting down gossamer node, pid=$GOSSAMER_PID...` ?

Reply:
whats wrong ?

Reply:
same as above, you use `...` when something is about to start but you do not here

Reply:
I see